rdr_service/dao/bq_questionnaire_dao.py

```diff
@@ -48,45 +101,90 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
+                    # for the first time, in case we need to confirm BQ schemas are in sync with RDR code table
+                    if ans.code_name not in fields.keys():
+                        logging.warning("""questionnaireResponseID {0} contains previously unrecognized answer code {1}
+                                for module {3}
+                            """.format(qr.questionnaire_response_id, ans.code_name, module_id))
+
+                    # Handle multi-select question codes (such as ethnicity or gender identity response options) where
+                    # user provided more than one answer.  Add to comma-separated list if there's already an answer val
+                    # This reproduces the GROUP_CONCAT SQL logic from the deprecated sp_get_questionnaire_answers proc
+                    if ans.code_name in ans_dict.keys() and ans_dict[ans.code_name]:
+                        prev_answer = ans_dict[ans.code_name]
+                        ans_dict[ans.code_name] = ",".join([prev_answer, ans.answer])
+                    else:
+                        ans_dict[ans.code_name] = ans.answer
+
+                fields.update(ans_dict)
+
+                # Merge the updated full survey fields dict with the response's metadata
+                data.update(fields)
+                qnans.append(data)
+
+            if len(qnans) == 0:
+                return None, list()
+
+            bqrs = list()
+            for qnan in qnans:
+                bqr = BQRecord(schema=table().get_schema(), data=qnan, convert_to_enum=convert_to_enum)
+                bqr.participant_id = p_id  # reset participant_id.
+
+                fields = bqr.get_fields()
+                for field in fields:
+                    fld_name = field['name']
+                    if fld_name in (
+                        'id',
+                        'created',
+                        'modified',
+                        'authored',
+                        'language',
+                        'participant_id',
+                        'questionnaire_response_id'
+                    ):
+                        continue
+
+                    fld_value = getattr(bqr, fld_name, None)
+                    if fld_value is None:  # Let empty strings pass.
+                        continue
+                    # question responses values need to be coerced to a String type.
+                    if isinstance(fld_value, (datetime.date, datetime.datetime)):
+                        setattr(bqr, fld_name, fld_value.isoformat())
+                    else:
+                        setattr(bqr, fld_name, str(fld_value))
+
+                    # Truncate zip codes to 3 digits
+                    if fld_name in ('StreetAddress_PIIZIP', 'EmploymentWorkAddress_ZipCode') and len(fld_value) > 2:
+                        setattr(bqr, fld_name, fld_value[:3])
+
+                bqrs.append(bqr)
+                if latest:
+                    break
```

Comment:
Since these are processed in order of oldest responses to newest, does passing in `latest` as True give the oldest response only?